spec/OpenLineage.json

```diff
@@ -64,6 +64,84 @@
         "schemaURL"
       ]
     },
+    "DatasetEvent": {
+      "type": "object",
+      "properties": {
+        "eventTime": {
+          "description": "the time the event occurred at",
+          "type": "string",
+          "format": "date-time"
+        },
+        "dataset": {
+          "allOf": [
+            { "$ref": "#/$defs/Dataset" }
+          ]
+        },
+        "producer": {
+          "description": "URI identifying the producer of this metadata. For example this could be a git url with a given tag or sha",
+          "type": "string",
+          "format": "uri",
+          "example": "https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/client"
+        },
+        "schemaURL": {
+          "description": "The JSON Pointer (https://tools.ietf.org/html/rfc6901) URL to the corresponding version of the schema definition for this RunEvent",
+          "type": "string",
+          "format": "uri",
+          "example": "https://openlineage.io/spec/2-0-0/OpenLineage.json#/$defs/DatasetEvent"
+        }
+      },
+      "required": [
+        "eventTime",
+        "dataset",
+        "producer",
+        "schemaURL"
```

Comment:
maybe eventTime, producer and schemaURL are part of a base `Event` type added with a AllOf like we do for Input/OutputDatasets and facets.

Reply:
This is cool! I will try it out.